### DISCUSSION TITLE
Is it exist the version of this library for C or C++ ?

### DISCUSSION THREAD

**Discussion by Ildaron** [hbldh/bleak#1309] [Q&A] [answered]
I implemented your library in python, works great, but very slowly with python, do you have sources for C ?

or in Python I do something wrong ? t is my script (this script lost many values from server)

```python
import asyncio
import sys
from bleak import BleakClient
import time

data_f = []
FIRST_NAME_ID = '0000fe42-8e22-4541-9d4c-21edae82ed19'
address = "00:80:E1:26:62:70"

async def main(address):
    async with BleakClient(address) as client:
        
        def callback(FIRST_NAME_ID, data):
            data_f.append(data[1])    
            #print (data[1])
            event.set()
            
            if len(data_f)==1000:
                with open("file.txt", "w") as output:
                    output.write(str(data_f))
                sys.exit()

        await client.start_notify(FIRST_NAME_ID, callback)
        print ("was connected")        
        while 1:
                         
          if not event.is_set():    
              await event.wait()  
          #await asyncio.sleep(0)
          event.clear()
          
event = asyncio.Event()
print('address:', address)
asyncio.run(main(address))
```

**Comment by dlech** [comment]
What operating system are you using?

Do you have logs of Bluetooth packets over the air vs Bluetooth packets received by the OS vs `file.txt` showing the data loss?

**Reply by Ildaron** [reply]
> What operating system are you using?
> 
> Do you have logs of Bluetooth packets over the air vs Bluetooth packets received by the OS vs `file.txt` showing the data loss?

Hi @dlech thank you for your reply,
Windows, 
I send from the Server (stm32) the next array = [0, count, byte_1, byte_2, bute_3],   where count = 1 + count (I used it is specially for test) , 250 times  per sec 
and receive the next data in the Python script in Windows 
thanks to count, I see that some of the data were lost 

it is example of data, that I use to check the BLE transfer 

zero, count,  byte_1, byte_2, bute_3
0 0 192 0 8
0 1 192 0 8
0 28 192 0 8
0 29 192 0 8
0 30 192 0 8
0 31 192 0 8
0 58 192 0 8
0 59 192 0 8
0 60 192 0 8
0 61 192 0 8
0 73 192 0 8
0 74 192 0 8
0 75 192 0 8
......
Do I understand correctly that python is too slow for real-time tasks?

**Reply by dlech** [reply, marked as answer]
> 250 times per sec

This would be once every 4 milliseconds. BLE connection intervals are generally something like every 30 or 60 milliseconds. HID devices might have a 15 ms interval. So there is no way the device could actually send notifications that frequently.

Using a Bluetooth sniffer will show you what is actually going over the air and when.

**Reply by Ildaron** [reply]
> > 250 times per sec
> 
> This would be once every 4 milliseconds. BLE connection intervals are generally something like every 30 or 60 milliseconds. HID devices might have a 15 ms interval. So there is no way the device could actually send notifications that frequently.
> 
> Using a Bluetooth sniffer will show you what is actually going over the air and when.

Thank you!!!! I missed it
